include/sys/vm_amap.h

```diff
@@ -0,0 +1,49 @@
+#ifndef _SYS_VM_AMAP_H_
+#define _SYS_VM_AMAP_H_
+
+#include <sys/types.h>
+#include <sys/vm.h>
+#include <stddef.h>
+
+typedef struct vm_amap vm_amap_t;
+typedef struct vm_aref vm_aref_t;
+
+struct vm_aref {
+  size_t offset;   /* offset in slots */
+  vm_amap_t *amap; /* underlying amap */
+};
+
+#define VM_AREF_EMPTY ((vm_aref_t){.offset = 0, .amap = NULL})
```

Comment:
Since this is used internally only in one place... do we need to have it defined in public interface?